src/ir/effects.h

```diff
@@ -143,6 +149,18 @@ class EffectAnalyzer {
   // or a continuation that is never continued, are examples of that.
   bool mayNotReturn = false;
 
+  // Since return calls return out of the body of the function before performing
+  // their call, they are indistinguishable from normal returns from the
+  // perspective of their surrounding code, and the return-callee's effects only
+  // become visible when considering the effects of the whole function
+  // containing the return call. To model this correctly, stash the callee's
+  // effects on the side and only merge them in after walking a full function
+  // body.
+  //
+  // We currently do this stashing only for the throw effect, but in principle
+  // we could do it for all effects if it made a difference.
```

Comment:
```suggestion
  // we could do it for all effects if it made a difference. (Only throw is noticeable
  // now because the only thing that can change between doing the call here
  // and doing it outside at the function exit is the scoping of try-catch blocks.
  // If future wasm scoping additions are added, we may need more here.)
```